### PR TITLE
feat: add left navigation menu for course

### LINKED ISSUE #53: SCORM xblock missing navigation menu
Hello,

I recently installed the SCORM xblock "https://github.com/overhangio/openedx-scorm-xblock" on my Open edX tutor Palm.

I tried to export a course from eXeLearning to SCORM 1.2 or 2004 format to upload to my Open edX.

When I upload the scorm package in Open edX I can only see the first page of the course. A negation menu is supposed to appear to move through all the sections of the course, or alternatively some buttons for "next" and "previous" but it doesn't.

Captura

I guess it is some xblock problem because I have uploaded the same package to another LMS platform and to scorm cloud and I can see the menu and navigation buttons.

Captura1

Any ideas or suggestions?

Thanks in advance. Best regards.

### PR DESCRIPTION
**Pull request by Danyal-Faheem**
Fixes #53.

**Changes**

- Added an option to enable navigation menu on the left side of the scorm panel.
- Navigation menu is built using the title tags of the items in the `imsmanifest.xml` file.
- Leaf item tags are also clickable in the navigation menu which change the current page to the clicked page.
- Added an option to specify width of navigation menu. Default size is 30% of scorm panel width, 70% is assigned to the course iframe.
- Added a `navigationClick` check on `GetValue` scorm API call as Chromium blocked synchronous calls on page unload and we didn't want to make an API call when changing pages using the navigation menu.

### CODE REVIEW

File: openedxscorm/scormxblock.py [hunk 1]
```diff
@@ -148,7 +148,23 @@ def scorm_storage(xblock):
         default=False,
         scope=Scope.settings,
     )
+    enable_navigation_menu = Boolean(
+        display_name=_("Display navigation menu"),
+        help=_(
+            "Select True to display a navigation menu on the left side to display table of contents"
+        ),
+        default=True,
```
Comment (ziafazal) [inline openedxscorm/scormxblock.py:156]: Can we set default to `False`

File: openedxscorm/scormxblock.py [hunk 2]
```diff
@@ -517,6 +538,8 @@ def update_package_fields(self):
         schemaversion = root.find(
             "{prefix}metadata/{prefix}schemaversion".format(prefix=prefix)
         )
+        
+        self.extract_navigation_titles(root, prefix)
```
Comment (ziafazal) [inline openedxscorm/scormxblock.py:542]: can if extract navigation only if `self.enable_navigation_menu` is set to `True`?

**Review by ziafazal — CHANGES_REQUESTED** [review]
@Danyal-Faheem overall looks good. I have couple of minor suggestions.

File: openedxscorm/scormxblock.py (hunk 1 above)
Comment (Danyal-Faheem) [reply openedxscorm/scormxblock.py:156]: Done!

File: openedxscorm/scormxblock.py (hunk 2 above)
Comment (Danyal-Faheem) [reply openedxscorm/scormxblock.py:542]: The problem with this was that we are only updating package_fields when we don't upload a new file in `studio_submit()`. Therefore, if a course author changes the `enable_navigation_menu` field after he has uploaded the zip file and submitted once, it won't go into the `update_package_fields` function at all and the titles won't be extracted.

**Review by ziafazal — APPROVED** [review]
👍

**Pull request merged**